### PR TITLE
Fix asyncio error when opening notebooks

### PR DESCRIPTION
**Pull request by dleen**
This is a fix for https://github.com/jupyter/notebook/issues/6164

`nest_asyncio` must be applied before any async tasks have been created
otherwise there could be tasks queued that are unpatched, and thus do
not respect nested loops. An example of an unpatched task can be seen in
the original issue:

```
<Task pending coro=<HTTP1ServerConnection._server_request_loop() running at /apps/python3/lib/python3.7/site-packages/tornado/http1connection.py:823>
```
which originates from Tornado.

A similar issue was reported in `nest-asyncio`: https://github.com/erdewit/nest_asyncio/issues/22
where the solution is to call `apply` on import so that unpatched tasks
do not get created.

### CODE REVIEW

File: notebook/notebookapp.py
Comment (kevin-bates) [inline notebook/notebookapp.py:2112]: Nice comment - thank you.

**Review by kevin-bates — APPROVED** [review]
These changes look good to me and I've confirmed the pending task error is not produced.  Thank you.

I'd prefer to wait for another review prior to merging and have requested the help of others.

**Review by blink1073 — APPROVED** [review]
Looks good, thank you @dleen!

**Comment by dleen** [comment]
@minrk Are you able to take a look?

**Comment by minrk** [comment]
Thanks!

**Pull request merged**
